doc/api/stream.md

```diff
@@ -271,68 +487,64 @@ readable.resume()
 readable.isPaused() // === false
 ```
 
-#### readable.pause()
+##### readable.pause()
 
 * Return: `this`
 
-This method will cause a stream in flowing mode to stop emitting
-[`'data'`][] events, switching out of flowing mode. Any data that becomes
-available will remain in the internal buffer.
+The `readable.pause()` method will cause a stream in flowing mode to stop
+emitting [`'data'`][] events, switching out of flowing mode. Any data that
+becomes available will remain in the internal buffer.
 
 ```js
-var readable = getReadableStreamSomehow();
+const readable = getReadableStreamSomehow();
 readable.on('data', (chunk) => {
-  console.log('got %d bytes of data', chunk.length);
+  console.log(`Received ${chunk.length} bytes of data.`);
   readable.pause();
-  console.log('there will be no more data for 1 second');
+  console.log('There will be no additional data for 1 second.');
   setTimeout(() => {
-    console.log('now data will start flowing again');
+    console.log('Now data will start flowing again.');
     readable.resume();
   }, 1000);
 });
 ```
 
-#### readable.pipe(destination[, options])
+##### readable.pipe(destination[, options])
 
 * `destination` {stream.Writable} The destination for writing data
 * `options` {Object} Pipe options
   * `end` {Boolean} End the writer when the reader ends. Default = `true`
 
-This method pulls all the data out of a readable stream, and writes it
-to the supplied destination, automatically managing the flow so that
-the destination is not overwhelmed by a fast readable stream.
+The `readable.pipe()` method attaches a [Writable][] stream to the `readable`,
+causing it to switch automatically into flowing mode and push all of its data
+to the attached [Writable][]. The flow of data will be automatically so that
+the destination Writable stream is not overwhelmed by a faster Readable stream.
```

Comment:
What happens when you pipe an `objectMode` stream into a non-`objectMode` stream? Or vice versa?